components/plan-build/bin/environment.sh

```diff
@@ -238,6 +238,54 @@ __populate_environment_from_deps() {
     fi
 }
 
+# For dependencies that do not have a RUNTIME_ENVIRONMENT file, we'll
+# at least assemble a PATH variable that can be used. To be complete
+# and utterly paranoid, we'll iterate through the TDEPS of the
+# dependency and assemble a PATH by layering all the PATHs of each of
+# those dependencies.
+#
+# Don't need to do this for packages that already have a
+# RUNTIME_ENVIRONMENT file, because this operation will have already
+# been done when those packages were built. This is to help preserve
+# backward compatibility.
+__assemble_legacy_path() {
+    local path_to_dep="${1}"
+    local path=""
+
+    warn "Assembling legacy path for ${path_to_dep}"
```

Comment:
These could probably be `build_line` instead since we've already warned and now we're doing the backwards compatible activity. Otherwise you start to accumulate a lot or `warns` which really only have one triggered source (the old package metadata).

Reply:
Sounds good; will change